src/dialogsearch.cpp

```diff
@@ -0,0 +1,172 @@
+#include "dialogsearch.h"
+#include "ui_dialogsearch.h"
+#include "mainwindow.h"
+#include "core.h"
+
+#include <data.h>
+#include <lmu_reader.h>
+#include <rpg_map.h>
+#include <functional>
+#include <vector>
+#include <tuple>
+
+DialogSearch::DialogSearch(QWidget *parent) :
+    QDialog(parent),
+    ui(new Ui::DialogSearch)
+{
+    ui->setupUi(this);
+
+    ui->list_result->setColumnCount(5);
+    ui->list_result->setRowCount(0);
+    ui->list_result->setSelectionBehavior(QAbstractItemView::SelectRows);
+    ui->list_result->setEditTriggers(QAbstractItemView::NoEditTriggers);
+}
+
+DialogSearch::~DialogSearch()
+{
+    delete ui;
+}
+
+void DialogSearch::updateUI()
+{
+    const QString format("%1: %2");
+
+    for (auto &v : Data::variables)
+        ui->combo_variable->addItem(format.arg(QString::number(v.ID), QString::fromStdString(v.name)));
+    for (auto &s : Data::switches)
+        ui->combo_switch->addItem(format.arg(QString::number(s.ID), QString::fromStdString(s.name)));
+    for (auto &i : Data::items)
+        ui->combo_item->addItem(format.arg(QString::number(i.ID), QString::fromStdString(i.name)));
+    for (auto &e : Data::commonevents)
+        ui->combo_eventname->addItem(format.arg(QString::number(e.ID), QString::fromStdString(e.name)));
+}
+
+void DialogSearch::on_button_search_clicked()
+{
+    objectData.clear();
+    ui->list_result->clear();
+    ui->list_result->setRowCount(0);
+    QStringList sl;
+    sl << "Map" << "Event" << "Event Page" << "Sourceline" << "Action";
+    ui->list_result->setHorizontalHeaderLabels(sl);
+
+
+
+    std::function<std::vector<std::tuple<int, int, int, int, std::vector<int>>>(const RPG::Map &)> func;
+
+
+    if (ui->radio_variable->isChecked())
+    {
+    }
+    else if (ui->radio_switch->isChecked())
+    {
+
+    }
+    else if (ui->radio_item->isChecked())
+    {
+        auto itemfunc = [](const int itemID, const RPG::Map& map){
+            std::vector<std::tuple<int, int, int, int, std::vector<int>>> res;
+            for (auto &e : map.events)
+                for (auto &p : e.pages)
+                    for (size_t line = 0; line < p.event_commands.size(); line++)
+                    {
+                        auto ev = p.event_commands[line];
+                        if (ev.code == RPG::EventCommand::Code::ChangeItems)
+                        {
+                            if (ev.parameters[2] == itemID)
+                                res.push_back({map.ID, e.ID, p.ID, line, ev.parameters});
+                        }
+                    }
+            return res;
+        };
+
+
+        QRegularExpression re("^(\\d+):?.*$");
+        auto itemID = re.match(ui->combo_item->currentText()).captured(1).toInt();
+        func = std::bind(itemfunc, itemID, std::placeholders::_1);
+    }
+    else //if (ui->radio_eventname->isChecked())
+    {
+
+    }
+
```

Comment:
This will crash if you don't use the item search parameter. Could you add here something like this while the other parameters get eventually implemented?:
```
if (!func) {
    QMessageBox::warning(this, "", "This search parameter isn't supported yet.");
    return;
}
```
It will require the \<QMessageBox\> header.